packages/variable-resolver/src/browser/variable-resolver-service.ts

```diff
@@ -101,16 +99,17 @@ export class VariableResolverService {
     }
 
     protected async doResolveString(value: string, context: VariableResolverService.Context): Promise<string> {
-        await this.resolveVariables(value, context);
-        return value.replace(VariableResolverService.VAR_REGEXP, (match: string, varName: string) => {
+        const variableRegExp = /\$\{(.*?)\}/g;
```

Comment:
static constant is better, why to move it here and break protected APIs?

Reply:
was fixed, please fetch again.
Now i use the static regexp but creates new object from it since it fails for me when i execute  parallel calls for this method (since the "exec" method is stateful)